projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringConnection.cs

```diff
@@ -1030,7 +1043,7 @@ private void StopRecoveryLoop()
         /// Handles commands when in the Recovering state.
         /// </summary>
         /// <param name="command"></param>
-        private async Task RecoveryLoopRecoveringHandler(RecoveryCommand command)
+        private void RecoveryLoopRecoveringHandler(RecoveryCommand command)
```

Comment:
Keep this async, and rather use a CancellationToken if we need a signal-breaker.